controller/displaycontroller.php

```diff
@@ -72,7 +72,34 @@ public function showPad($file, $dir) {
             'url' => $url,
             'title' => $title,
         ];
-        $response = new TemplateResponse($this->appName, 'viewer', $params, 'blank');
+
+        // Check for valid URL
+        // Get File-Ending
+        $split = explode(".", $file);
+        $fileending = $split[count($split)-1];
+
+        // Get Host-URL
+        if($fileending === "calc") {
+            $host = \OCP\Config::getAppValue('ownpad', 'ownpad_ethercalc_host', false);
+        }
+        elseif($fileending === "pad") {
+            $host = \OCP\Config::getAppValue('ownpad', 'ownpad_etherpad_host', false);
+        }
+
+        // Escape all Slashes in URL to use in RegEx
+        $hostreg = preg_replace("/\//", "\/", $host);
```

Comment:
It would be better to use [`preg_quote()`](http://php.net/manual/en/function.preg-quote.php) here to be sure all regular expression characters are properly escaped.